common/persistence/sql/sqlExecutionStore.go

```diff
@@ -740,9 +740,20 @@ func (m *sqlExecutionStore) GetTransferTasks(
 	ctx context.Context,
 	request *p.GetTransferTasksRequest,
 ) (*p.GetTransferTasksResponse, error) {
-
+	minReadLevel := request.ReadLevel
+	if len(request.NextPageToken) > 0 {
+		readLevel, err := deserializePageToken(request.NextPageToken)
+		if err != nil {
+			return nil, convertCommonErrors(m.db, "GetTransferTasks", "failed to deserialize page token", err)
+		}
+		minReadLevel = readLevel
+	}
 	rows, err := m.db.SelectFromTransferTasks(ctx, &sqlplugin.TransferTasksFilter{
-		ShardID: m.shardID, MinTaskID: &request.ReadLevel, MaxTaskID: &request.MaxReadLevel})
+		ShardID:   m.shardID,
+		MinTaskID: minReadLevel,
+		MaxTaskID: request.MaxReadLevel,
+		PageSize:  request.BatchSize,
```

Comment:
There are 2 upper bound constraints here. Did you check if this works fine? I'm wondering if it would be a problem when the last row's taskID < maxReadLevel (i.e. if there's any component in the stack depending on that logic)

Even if this is fine, it would be useful to write a unittest to make sure this scenario won't be broken in the future.

Reply:
We already have test cases.

Reply:
see https://github.com/uber/cadence/blob/master/common/persistence/persistence-tests/persistenceTestBase.go#L1274